### ISSUE TITLE
Using :file parameter in org-bable doesn't work.

### ISSUE THREAD

**Issue by josteink** [josteink/wsd-mode#20]
Long story short, a file with `file-name` is written, which contains the text `file-name`.

Hardly useful :)

**Note:** Omitting file-parameter and everything works fine.

**Edit:** Demo-document below.

````org
* org-document

#+BEGIN_SRC wsd
title Does work
#+END_SRC

#+RESULTS:
[[file:/tmp/wsd_3454353.png]]

#+BEGIN_SRC wsd :file test.png
title Does NOT work
#+END_SRC

#+RESULTS:
[[file:test.png]]

````

**Comment by josteink** [comment]
Single-stepping through [the code](https://github.com/josteink/wsd-mode/blob/master/ob-wsdmode.el#L36-L54) can see the exact same code-execution with and without the `:file` parameter. The code does the same and executes correctly.

This leads us to the conclusion that `org-mode` itself treats our `org-babel-execute`-function differently based on context, and that we need to account for this:

* Without a `:file` parameter, output is assumed to be a file and used as such. Returning the file-name in this case works.
* With a `:file` parameter, output is assumed to be contents to be written to the specified file. Returning the file-name means we will be writing the file-name to the file. Which, if true, is not very useful at all.

If this is correct, there's at least three different fixes:

* Change the parameter used to be called something not `:file` (although this breaks "compatibility" with `plantuml-mode`, but that compatibility is nowhere near 100% anyway).
* Introduce duplicate code-paths. When we have a `file:`-parameter read the contents of the file and return that instead of the file-name.
* Try to alter the org-state so that `org-babel` always assumes that the return-value is a file-name.

Personally I think the first option sounds like the simplest fix, the last option a decent second-best fix and ode duplication should be avoided at all cost.

**Comment by wasamasa** [comment]
I can verify that the broken example writes "test.png" to `test.png`.

**Comment by wasamasa** [comment]
OK, I've figured it out.  If you step through the babel execution code, you'll notice that at some point the image has been written out successfully to disk, but is later overwritten.  Therefore, I've searched Org's sources and found out that `:file` is special-cased: If it's specified, the argument is interpreted as the file name *and* any non-nil output is written to that file name.  So, the solution is simple, either return `img-file` or `nil`, depending on whether `out-file` is set or not.
